asterisk/agi/library/Agi/Action/ServiceAction.php

```diff
@@ -114,4 +117,45 @@ protected function _processGroupPickUp()
 
     }
 
+    protected function _processRecordLocution()
+    {
+        // Local variables to improve readability
+        $service = $this->_service;
+        $caller = $this->_caller;
+
+        // Get Locution id to be recorded
+        $locutionId = substr($this->agi->getExtension(), strlen($service->getCode()) + 1);
```

Comment:
I would extract `$this->agi->getExtension()` to a new variable with a comment describing it's format:  "*" . serviceCode() . modelId